plugin/terminaldocument.vim

```diff
@@ -6,13 +6,18 @@ autocmd BufEnter,BufFilePost * call s:update_document(0)
 " these events might change the nature of the file,
 " so we try to refresh its icon
 autocmd BufWritePost,FileChangedShellPost,ShellCmdPost * call s:update_document(1)
-autocmd VimLeave * call s:set_osc6('')
+
+if &term != 'xterm-ghostty'
+  " For ghostty setting this to empty results in the path being set to "/"
+  " At least before https://github.com/ghostty-org/ghostty/pull/3399
+  autocmd VimLeave * call s:set_osc('')
+endif
 
 function! s:update_document(refresh)
   let path = expand('%:p')
-  if empty(path) || !filereadable(path)
+  if empty(path)
```

Comment:
As I recall the `filereadable()` test is necessary because some plugins create virtual buffers with random names. I tried checking for `buftype`, but the `BufEnter` event occurs before e.g. NERDTree gets a chance to change it. Did you find any scenario where this was undesirable?

Reply:
I was trying to handle the case where if I do `vim foo.html` it will have the correct file type icon. It does seem to not work with even some of the built-in special buffers, so I'll revert that bit.